### PR TITLE
[kf5auth] new port

### PR DESCRIPTION
**Pull request by wrobelda**
**Describe the pull request**
Adds new kf5auth port. Depends on https://github.com/microsoft/vcpkg/pull/17268 

- Which triplets are supported/not supported? Have you updated the CI baseline?
All official should be supported.

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
Yes

### CODE REVIEW

File: ports/kf5auth/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,36 @@
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO KDE/kauth
+    REF v5.75.0
```
Comment (NancyLi1013) [inline ports/kf5auth/portfile.cmake:4]: It seems that this is not the latest release. Can we update it to the latest?

File: ports/kf5auth/vcpkg.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
+{
+  "name": "kf5auth",
+  "version-string": "5.75.0",
```
Comment (NancyLi1013) [inline ports/kf5auth/vcpkg.json:3]:
```suggestion
  "version": "5.75.0",
```
We prefer to use more proper name to describe this field. Please see https://github.com/microsoft/vcpkg/blob/master/docs/specifications/versioning.md#22-package-versions

File: versions/k-/kf5auth.json [hunk 3]
```diff
@@ -0,0 +1,9 @@
+{
+  "versions": [
+    {
+      "git-tree": "c1bd3c294a0bd9d60b9f5bee1548d614c8185c49",
+      "version-string": "5.75.0",
```
Comment (NancyLi1013) [inline versions/k-/kf5auth.json:5]:
```suggestion
      "version": "5.75.0",
```
The same as above.

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

File: ports/kf5auth/vcpkg.json (hunk 2 above)
Comment (wrobelda) [reply ports/kf5auth/vcpkg.json:3]: Yes, I am actually aware – I was researching wether we're compliant with SymVer, I'll update accordingly.

File: ports/kf5auth/portfile.cmake (hunk 1 above)
Comment (wrobelda) [reply ports/kf5auth/portfile.cmake:4]: Indeed. Once #13467 gets merged, I will follow up with a new PR to update all the ports included in it. Once that one gets accepted, I'll continue to add new additional ports, such us `kf5auth` in this PR.

**Comment by NancyLi1013** [comment]
Wow, Seems you need to rebase this PR now, many changes were brought in this PR. Or you can open a new pull request to adding this port.

**Comment by wrobelda** [comment]
> Wow, Seems you need to rebase this PR now, many changes were brought in this PR. Or you can open a new pull request to adding this port.

Indeed, will be rebasing, but still waiting for https://github.com/microsoft/vcpkg/pull/17268 to get merged.

**Comment by wrobelda** [comment]
@NancyLi1013 #17268 got merged. Rebased and ready to go. The static build failure seems to be due to intermittent CI/CD connectivity issue.

File: ports/kf5auth/vcpkg.json [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+  "version": "5.81.0",
+  "description": "Execute actions as privileged user",
+  "homepage": "https://api.kde.org/frameworks/kauth/html/index.html",
+  "dependencies": [
+    "ecm",
+    "kf5coreaddons",
+    "qt5-base",
+    "qt5-tools"
+  ]
+}
```
Comment (NancyLi1013) [inline ports/kf5auth/vcpkg.json:12]: If we use these new functions, we should also add `vcpkg-cmake` and `vcpkg-cmake-config` to the dependencies as follows:

```
   "dependencies": [
    {
      "name": "vcpkg-cmake",
      "host": true
    },
    {
      "name": "vcpkg-cmake-config",
      "host": true
    }
```

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Comment by wrobelda** [comment]
@NancyLi1013 all fixed, thanks for the instructions.

**Review by NancyLi1013 — APPROVED** [review]

**Comment by NancyLi1013** [comment]
LGTM now, thanks for adding this port. @wrobelda

**Comment by strega-nil-ms** [comment]
Cool! Thanks @wrobelda :)

**Pull request merged**